### PR TITLE
Remove checkbox and radio borders

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

Removes the checkbox and radio borders. Amends the new hover effect slightly to accommodate this.

| new | base |
|--------|--------|
| Screenshot 2024-09-24 at 18 32 15 | Screenshot 2024-09-24 at 18 32 36 | 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by aliabid94** [comment]
Can we also do this through theme variables rather than removing the variables entirely? So that some themes can preserve them and others can remove them.

**Comment by pngwn** [comment]
Adding this comment I made from slack. 

TLDR: I don't think the borders are the real problem, even ignoring this PR there are other issues with things looking too bare, mainly because if the removal of the separators with no other adjustments. 

---
**Chatbot with radios** 

image

**blocks_essay demo**

image

**kitchen sink**
image

Observations:

- chatbot looks absurd but it always did so that isn't a big deal. 
- blocks_essay looks great
- kitchen sink looks naked (only way i can describe it)

Possible solutions:

- So the reason i think the chatbot looks silly is cos what we should try to do is balance the row + align the radio button.
- I think the blocks essay might need more padding margin above
- Not sure about kitchen sink, the checkboxes actually look okay but dno about the radios, they seem more problematic

I think radios might need more padding between them too actually, just a touch. 

Yeah they do, there can't be the same space between the label text + the previous radio or the relationship isn't clear, checkboxes don't seem to have that issue


hmmm, i know we removed the separator because it was noisy but i feel the labels + margins aren't pulling their weight in separating things out

I think the label text needs to be more prominent and maybe we need more margin when stacking form elements like this. The ones with boxes look okay but slider etc look a bit lost + floaty

I feel like the label should be more prominent than the actual value text for example, those are like mini headings and are the only thing that really dictate the hierarchy now. 

could also just be larger instead or darker, probably not both. 

or maybe bolder but may look bad in dark mode unless we drop the weight by 100

**Comment by abidlabs** [comment]
tbh I prefer the borders in all of three demos above

**Comment by pngwn** [comment]
Depends what they are there for, in a larger form i think they look really noisy. In small doses they look okay (but i don't think they look great).

**Comment by abidlabs** [comment]
Closing as the consensus is that we'll stick with the borders
